# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/ParentProcessDocumentChannel.cpp
# Commit: 1177913e1edf
# Full Hash: 1177913e1edf443da73d438f95a334e466b623e2
# Author: Manuel Bucher <manuel@mozilla.com>
# Date: 2022-12-02 21:26:33
# Regressor Bug: 1771867
# File Overlap Count: 1
# Description:
#   Bug 1771867 - Early Hints Phase 2 - Part 8: Add list of open channels from OngoingEarlyHints to RedirectToRealChannelArgs r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161179
# ==============================================================================

diff -r d33ccbbf407d -r 1177913e1edf netwerk/ipc/ParentProcessDocumentChannel.cpp
--- a/netwerk/ipc/ParentProcessDocumentChannel.cpp	Fri Dec 02 09:45:26 2022 +0000
+++ b/netwerk/ipc/ParentProcessDocumentChannel.cpp	Fri Dec 02 09:45:27 2022 +0000
@@ -48,7 +48,8 @@
 ParentProcessDocumentChannel::RedirectToRealChannel(
     nsTArray<ipc::Endpoint<extensions::PStreamFilterParent>>&&
         aStreamFilterEndpoints,
-    uint32_t aRedirectFlags, uint32_t aLoadFlags) {
+    uint32_t aRedirectFlags, uint32_t aLoadFlags,
+    const nsTArray<EarlyHintConnectArgs>& aEarlyHints) {
   LOG(("ParentProcessDocumentChannel RedirectToRealChannel [this=%p]", this));
   nsCOMPtr<nsIChannel> channel = mDocumentLoadListener->GetChannel();
   channel->SetLoadFlags(aLoadFlags);
@@ -207,7 +208,8 @@
         RefPtr<RedirectToRealChannelPromise> p =
             self->RedirectToRealChannel(
                     std::move(aResolveValue.mStreamFilterEndpoints),
-                    aResolveValue.mRedirectFlags, aResolveValue.mLoadFlags)
+                    aResolveValue.mRedirectFlags, aResolveValue.mLoadFlags,
+                    aResolveValue.mEarlyHints)
                 ->Then(
                     GetCurrentSerialEventTarget(), __func__,
                     [self](RedirectToRealChannelPromise::ResolveOrRejectValue&&